### PR TITLE
[Filebeat] Replace Suricata/Eve fields with aliases to ECS fields

### PR DESCRIPTION
**Pull request by adriansr**
This PR avoids duplicate data in documents ingested via the Suricata Eve fileset  by replacing a few fields with aliases to ECS fields.

This allows to maintain the full set of fields Suricata users may expect while at the same time reducing the size of the events.

The aliased fields are:

| Aliased field | Target ECS field |
|-----------------|------------------------|
| suricata.eve.alert.action | event.outcome |
| suricata.eve.alert.severity | event.severity |
| suricata.eve.app_proto | network.protocol |
| suricata.eve.dest_ip | destination.ip |
| suricata.eve.dest_port | destination.port |
| suricata.eve.event_type | event.type |
| suricata.eve.fileinfo.filename | file.path |
| suricata.eve.fileinfo.size | file.size |
| suricata.eve.flow.start | event.start |
| suricata.eve.flow.bytes_toclient | destination.bytes |
| suricata.eve.flow.bytes_toserver | source.bytes |
| suricata.eve.flow.pkts_toclient | destination.packets |
| suricata.eve.flow.pkts_toserver | source.packets |
| suricata.eve.http.hostname | url.domain |
| suricata.eve.http.http_method | http.request.method |
| suricata.eve.http.http_refer | http.request.referrer |
| suricata.eve.http.http_user_agent | user_agent.original |
| suricata.eve.http.length | http.response.body.bytes |
| suricata.eve.http.status | http.response.status_code |
| suricata.eve.proto | network.transport |
| suricata.eve.src_ip | source.ip |
| suricata.eve.src_port | source.port |
| suricata.eve.timestamp | `@timestamp` |

Also, the following non-related fixes are performed:
- "http.response.status_code" was being set from a string and not an integer.
- "user_agent.original" was being inadvertedly removed by the user_agent processor.
- Reformatted the ingest pipeline with standard JSON formatting.

### CODE REVIEW

**Comment by adriansr** [comment]
Note to reviewers: To review the changes to the ingest pipeline is easier if you look at the changes added in every individual commit in the PR.

File: x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json [hunk 1]
```diff
@@ -76,6 +58,7 @@
         "user_agent.major": "7",
         "user_agent.minor": "58",
         "user_agent.name": "curl",
+        "user_agent.original": "curl/7.58.0",
```
Comment (webmat) [inline x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json]: Good catch!

File: x-pack/filebeat/module/suricata/eve/ingest/pipeline.json [hunk 2]
```diff
@@ -317,6 +306,38 @@
...
+                    "suricata.eve.http.http_user_agent",
+                    "suricata.eve.http.length",
+                    "suricata.eve.http.status",
+                    "suricata.eve.proto",
+                    "suricata.eve.src_ip",
+                    "suricata.eve.src_port",
+                    "suricata.eve.timestamp"
+                ],
+                "ignore_missing": true
+            }
```
Comment (webmat) [inline x-pack/filebeat/module/suricata/eve/ingest/pipeline.json]: For most of these fields, you could actually get rid of the `convert { type: string}` and do a straight field rename. Would look much more like the other modules, and would remove the need to have this huge `remove` operation.

File: x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json [hunk 3]
```diff
@@ -100,29 +81,18 @@
...
-        "suricata.eve.event_type": "http",
         "suricata.eve.flow_id": 2115002772430095,
-        "suricata.eve.http.hostname": "192.168.86.28",
         "suricata.eve.http.http_content_type": "text/xml",
-        "suricata.eve.http.http_method": "GET",
-        "suricata.eve.http.http_user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36",
-        "suricata.eve.http.length": 1155,
         "suricata.eve.http.protocol": "HTTP/1.1",
-        "suricata.eve.http.status": 200,
         "suricata.eve.http.url": "/dd.xml",
```
Comment (webmat) [inline x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json]: suricata.eve.http.url => url.path

File: x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json [hunk 4]
```diff
@@ -38,34 +38,16 @@
...
-        "suricata.eve.dest_ip": "93.184.216.34",
-        "suricata.eve.dest_port": 80,
-        "suricata.eve.event_type": "alert",
-        "suricata.eve.flow.bytes_toclient": 1654,
-        "suricata.eve.flow.bytes_toserver": 347,
-        "suricata.eve.flow.pkts_toclient": 3,
-        "suricata.eve.flow.pkts_toserver": 4,
         "suricata.eve.flow_id": 2191386088856669,
-        "suricata.eve.http.hostname": "example.net",
         "suricata.eve.http.http_content_type": "text/html",
```
Comment (ruflin) [inline x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json]: @webmat This and the one below could be something for ECS in the future.

**Comment by ruflin** [comment]
I initially thought these should not go into the ecs-migration.yml file but now I think it would make sense to add them there. This has two advantages:

* In case you missed renaming one of the fields above in the dashboards, the migration script will catch it.
* We can show a full list of files that we migrated including this out of ecs-migration.yml file.

File: x-pack/filebeat/module/suricata/eve/ingest/pipeline.json (hunk 2 above)
Comment (adriansr) [reply x-pack/filebeat/module/suricata/eve/ingest/pipeline.json]: You are completely right. There's a lot of cleanup that can be done

File: x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json (hunk 3 above)
Comment (adriansr) [reply x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json]: I'm not so sure about that one, as we will be losing information if s.e.http.url has query and/or fragment part (those go into url.query and url.fragment).

**Comment by adriansr** [comment]
@ruflin I've updated the ecs-migration.yml, please have a look. I'm unsure if I used the right set of flags. Is there a way to test this migration script?

**Comment by ruflin** [comment]
@adriansr The migration script for the dashboards can be found here: https://github.com/elastic/beats/pull/9998 I still have to create the script to generate a nice changelog entry.

File: x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json (hunk 3 above)
Comment (ruflin) [reply x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json]: perhaps then url.original?

Comment (webmat) [reply x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json]: Yeah, in this case `url.original` should be used. This is for the value as observed.

Then optionally (out of scope here) the other fields could be populated to break it down (url.path, url.query, url.fragment), or rebuild full context (url.full).

File: x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json (hunk 4 above)
Comment (webmat) [reply x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json]: Yeah I took a note to add `content_type`, when reviewing this PR, indeed :-)

What's the other field you're referring to? The field below is `method` LOL. And `.hostname` => `.domain` ;-)

File: x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json (hunk 3 above)
Comment (adriansr) [reply x-pack/filebeat/module/suricata/eve/test/eve-small.log-expected.json]: Oh, I overlooked url.original. Done!

File: x-pack/filebeat/module/suricata/eve/ingest/pipeline.json [hunk 5]
```diff
@@ -76,14 +77,6 @@
...
-            "convert": {
-                "field": "suricata.eve.http.hostname",
-                "target_field": "destination.domain",
-                "type": "string",
-                "ignore_missing": true
-            }
-        },
         {
             "convert": {
                 "field": "suricata.eve.http.http_refer",
```
Comment (webmat) [inline x-pack/filebeat/module/suricata/eve/ingest/pipeline.json]: This one's straightforward as well. Any reason why you're not renaming it?

Comment (adriansr) [reply x-pack/filebeat/module/suricata/eve/ingest/pipeline.json]: Nope, just skipped it by mistake. Good catch

**Comment by adriansr** [comment]
jenkins, test this

**Pull request merged**